Source/Pragmatic/Interaction/QueryExecutorExtensions.cs

```diff
@@ -29,6 +29,14 @@ public static Option<object> GetById(this QueryExecutor queryExecutor, Type enti
             Argument.IsNotNull(criteria, "criteria");
 
             return queryExecutor.Execute(new GetOneQuery<T> { Criteria = criteria });
+        }
+
+        public static Option<T> GetOne<T>(this QueryExecutor queryExecutor, Expression<Func<T, bool>> criteria, OrderBy<T> orderBy) where T : class
+        {
+            Argument.IsNotNull(queryExecutor, "queryExecutor");
+            Argument.IsNotNull(criteria, "criteria");
+

```

Comment:
@hudo Thanks for the extension and the bug fix in the GetOneQuery! That one slipped unnoticed so far ;-)

Please add the precondition here for the orderBy argument (IsNotNull) and let's pull the request afterwards.

Thanks!